src/server.rs

```diff
@@ -563,34 +578,261 @@ impl ServerState {
+                    assert!(sequence == latest_processed_seq_num);
+                    let _response_packet = self.process_player_request_action(player_id, action);
+                }
+                _ => panic!("Development bug: Non-response packet found in client buffered RX queue")
+            }
+        }
+    }
+
+    fn process_player_buffered_packets(&mut self, players_to_update: &Vec<PlayerID>) {
+        for player_id in players_to_update {
+            self.process_queued_rx_packets(*player_id);
+        }
+    }
+
+    fn process_buffered_packets_in_lobby(&mut self){
+        let mut players_to_update: Vec<PlayerID> = vec![];
+
+        if self.players.len() == 0 {
+            return;
+        }
+
+        // Skip players if they're in a game
+        for player in self.players.values() {
+            if player.game_info.is_some() {
+                continue;
+            }
+            players_to_update.push(player.player_id);
+        }
+
+        if players_to_update.len() != 0 {
+            self.process_player_buffered_packets(&players_to_update);
+        }
+    }
+
+    fn resend_expired_tx_packets(&mut self, udp_tx: &mpsc::UnboundedSender<(SocketAddr, Packet)>) {
+        let mut players_to_update: Vec<PlayerID> = vec![];
+
+        if self.players.len() == 0 {
+            return;
+        }
+
+        for player in self.players.values() {
+            players_to_update.push(player.player_id);
+        }
+
+        if players_to_update.len() != 0 {
+            for player_id in players_to_update {
+                // If any processed packets result in responses, prepare them below for transmission
+                let player_addr: SocketAddr = self.get_player_mut(player_id).addr;
+                let ack = self.get_player_mut(player_id).request_ack;
+
+                let player_network: Option<&mut NetworkManager> = self.network_map.get_mut(&player_id);
+                if let Some(player_net) = player_network {
+                    if player_net.tx_packets.len() == 0 {
+                        continue;
+                    }
+
+                    let indices = player_net.tx_packets.get_retransmit_indices();
+                    trace!("[Sending expired responses to client from TX Buffer]: {:?} Len: {}", player_id, indices.len());
+                    player_net.retransmit_expired_tx_packets(udp_tx, player_addr, ack, &indices);
+
+                } else {
+                    error!("I haven't found a NetworkManager for Player: {}", player_id);
+                    continue;
+                }
+            }
+        }
+    }
+
+    fn process_buffered_packets_in_rooms(&mut self) {
+        let mut players_to_update: Vec<PlayerID> = vec![];
+
+        if self.rooms.len() == 0 {
+            return;
+        }
+
+        for room in self.rooms.values() {
+            if room.player_ids.len() == 0 {
+                continue;
+            }
+
+            for player_id in &room.player_ids {
```

Comment:
Same suggestion as in other file: prefer `for &player_id in ...` and then reference with `player_id` instead of `*player_id`.